### ISSUE TITLE
Taskbar icons eventually start disappearing after a day or two of running

### ISSUE THREAD

**Issue by kode54** [LBCrion/sfwbar#194]
Furthermore, restarting sfwbar, the icons only appear on one of my monitors. It could also be exacerbated by the fact I have two different scale factors set.

I have an LG 24UD58-B, 3840x2160@60Hz, set to 2.0 scale, and a Dell P2414 1920x1080@60Hz, set to 1.0 scale. I'm running labwc as my compositor, using wlroots master (at 0.18.0 unreleased version).

**Comment by LBCrion** [comment]
Are you using the latest version of sfwbar? Icons only appearing on one monitor sounds like a symptom of a bug fixed in  e4da910.

Icons disappearing would be harder to track. Do all icons start disappearing? (i.e. window icons, module icons, etc?).

**Comment by kode54** [comment]
Only application icons are disappearing from the taskbar buttons. It seems related to monitor restore from a hotplug event on DPMS on, and doesn’t seem to affect buttons that appear after the icons vanish in the first place. I’m currently running:

sfwbar-git r1471.573aeb2-1

**Comment by LBCrion** [comment]
I think I found the cause. Would you mind trying the latest git version to
see if it fixes the issue?

**Comment by kode54** [comment]
Not sure. I just logged in to my desktop with the new build (`r1477.0352267-1`) and the icons are missing on the taskbar on my secondary (1.0 scale) monitor. The notification area icons are there, but the taskbar just has buttons with bare labels and no icons.

**Comment by LBCrion** [comment]
What config are you using? Does your taskbar use grouping?

**Comment by kode54** [comment]
I’m using the default sfwbar.config, minus three changes:

1) Change terminal to foot
2) Comment out network-module widget because I’m not using NetworkManager
3) Comment out sway-lang widget because I’m not using Sway

**Comment by LBCrion** [comment]
Ok, I think I found the cause (again). Would you mind giving the latest git
version a go?

**Comment by kode54** [comment]
Seems to work sometimes. Just logged into a Wayfire session now. All icons are present, except for Wezterm's icon on the secondary monitor:

![image](https://github.com/LBCrion/sfwbar/assets/796316/4e5f3ca5-b63d-4fc1-8f32-6be0341dc20c)

**Comment by kode54** [comment]
Holy heck, sorry about that, but thanks, that seems to have fixed it on startup, and I'll report back in a day or two whether they manage to keep their icons. I'm thinking they will this time, though.

**Comment by LBCrion** [comment]
Just checking if you've encountered this issue since the last fix went in?

**Comment by kode54** [comment]
No, but I did randomly encounter a different one that I thought had been fixed. I'll track down the issue to post there.
